python/sglang/srt/grpc_server.py

```diff
@@ -0,0 +1,74 @@
+import traceback
+from typing import Any, AsyncGenerator, Callable, Dict
+
+import grpc
+
+from sglang.srt.managers.io_struct import GenerateReqInput
+from sglang.srt.proto import completion_pb2, completion_pb2_grpc
+
+
+class CompletionServicer(completion_pb2_grpc.CompletionServiceServicer):
+    def __init__(
+        self,
+        generate_request: Callable[
+            [GenerateReqInput], AsyncGenerator[Dict[str, Any], None]
+        ],
+    ):
+        self.generate_request = generate_request
+
+    async def Complete(
+        self,
+        request: completion_pb2.CompletionRequest,
+        context: grpc.aio.ServicerContext,
+    ) -> AsyncGenerator[completion_pb2.CompletionResponse, None]:
+        try:
+            # Convert gRPC request to internal format
+            adapted_request = GenerateReqInput(
+                text=request.prompt,
+                sampling_params={
+                    "max_new_tokens": request.max_tokens,
+                    "temperature": request.temperature,
+                    "top_p": request.top_p,
+                    "top_k": request.top_k,
+                    "min_p": request.min_p,
+                    "frequency_penalty": request.frequency_penalty,
+                    "presence_penalty": request.presence_penalty,
+                    "stop": list(request.stop),
+                    "ignore_eos": request.ignore_eos,
+                },
+                stream=request.stream,
+            )
+
+            # Process request through tokenizer manager
+            async for content in self.generate_request(adapted_request):
```

Comment:
we need to hanadle the client closing the connection and abort the request (I think we can do it using context object). We can also do it as a follow up, and in that case, can you create a separate isuse?